perceval/backends/core/gitter.py

```diff
@@ -0,0 +1,378 @@
+        """
+        search_fields = {
+            DEFAULT_SEARCH_FIELD: self.metadata_id(item),
+            'group': self.group,
+            'room': self.room,
+            'room_id': self.room_id
+
+        }
+
+        return search_fields
+
+    def fetch(self, category=CATEGORY_MESSAGE, from_date=DEFAULT_DATETIME):
+        """Fetch the messages from the room.
+
+        This method fetches the messages sent in the room that were
+        sent since the given date.
+
+        :param category: the category of items to fetch
+        :param from_date: date from which messages are to be fetched
+
+        :returns: a generator of messages
+        """
+        if not from_date:
+            from_date = DEFAULT_DATETIME
+
+        from_date = from_date.strftime('%Y-%m-%d')
+        kwargs = {
+            'from_date': from_date,
+        }
+
+        items = super().fetch(category, **kwargs)
+
+        return items
+
+    def fetch_items(self, category, **kwargs):
+        """Fetch the messages.
+
+        :param category: the category of items to fetch
+        :param kwargs: backend arguments
+
+        :returns: a generator of items
+        """
+        from_date = kwargs['from_date']
+
+        logger.debug("Get Gitter message paginated items of room: %s from date: %s", self.room, from_date)
+
+        from_date = str_to_datetime(from_date).timestamp()
+        fetching = True
+        before_id = None
+        num_msgs = 0
+        page = 0
+        room = urijoin(self.group, self.room)
+
+        self.room_id = self.client.get_room_id(room)
+
+        if not self.room_id:
+            msg = "Room id not found for room %s" % room
+            logger.error(msg)
+
+            raise BackendError(cause=msg)
+
+        logger.debug("Room id of room: %s is %s", self.room, self.room_id)
+
+        while fetching:
+            page += 1
+            logger.debug("Page: %i" % page)
+            message_group = self.client.message_page(self.room_id, before_id)
+
+            message_group = message_group.json()
+
+            if not message_group:
+                fetching = False
+                continue
+
+            for raw_message in message_group:
+
+                if str_to_datetime(raw_message['sent']).timestamp() > from_date:
+                    num_msgs += 1
+                    yield raw_message
+                else:
+                    fetching = False
+
+            before_id = message_group[0]['id']
+
+        logger.debug("Fetch process completed: %s messages fetched", num_msgs)
+
+    @classmethod
+    def has_archiving(cls):
+        """Returns whether it supports archiving items on the fetch process.
+
+        :returns: this backend supports items archive
+        """
+        return True
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend does not support items resuming
+        """
+        return True
```

Comment:
I have been playing with the backend, and the messages returned are from the most recent to the last recent. In this case, the backend doesn't support resuming, because in case of failure the fetching process must start over.

The resuming is supported when the data is retrieved from the oldest to the most recent, because in case of failures the fetching process can start from the last item collected.

Reply:
It seems that it isn't possible to fetch the messages in ASC order. If you find something, please let me know.

Reply:
Thank you for the info :)
I rechecked the Gitter API doc. The API does not support fetching the messages in ascending order. I will make this change :)